core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -2049,12 +2054,31 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
     * res0: Unit = ()
     * }}}
     */
-  def parEvalMapUnordered[F2[x] >: F[
-    x
-  ]: Concurrent, O2](
+  def parEvalMapUnordered[F2[x] >: F[x]: Concurrent, O2](
       maxConcurrent: Int
   )(f: O => F2[O2]): Stream[F2, O2] =
-    map(o => Stream.eval(f(o))).parJoin(maxConcurrent)
+    Stream
+      .eval {
+        (
+          Semaphore[F2](maxConcurrent),
+          Queue.bounded[F2, Option[O2]](1)
+        ).tupled
+      }
+      .flatMap { case (semaphore, queue) =>
```

Comment:
Recently I have rewritten method code to use an `Stream.force` and then a main for-comprehension in `f`: 

```Scala
     )(f: O => F2[O2]): Stream[F2, O2] = Stream.force {
     for {
        semaphore <- Semaphore[F2](maxConcurrent)
        queue         <- Queue.bounded[F2, Option[O2]](1)
     } yield {
          /* --- the stream here.*/
     }
```

Reply:
Thanks, TIL `Stream.force` :)))
I've looked at `force`, I'm not trying to introduce `F[Stream[F, O]]` into `Stream`, thus I think `.eval(..).flatMap` is the right combination, or am I missing something?

Reply:
ohhh... I think I got it! I'll try it, thnx!